src/telemetry/telemetry.c

```diff
@@ -264,6 +269,259 @@ add_job_counts(JsonbParseState *state)
 	ts_jsonb_add_int32(state, REQ_NUM_USER_DEFINED_ACTIONS_FIXED, counts.user_defined_action_fixed);
 }
 
+static JsonbValue *
+add_errors_by_sqlerrcode_internal(JsonbParseState *parse_state, const char *job_type,
+								  Jsonb *sqlerrs_jsonb)
+{
+	JsonbIterator *it;
+	JsonbIteratorToken type;
+	JsonbValue val;
+	JsonbValue *ret;
+	JsonbValue key = {
+		.type = jbvString,
+		.val.string.val = pstrdup(job_type),
+		.val.string.len = strlen(job_type),
+	};
+
+	ret = pushJsonbValue(&parse_state, WJB_KEY, &key);
+	ret = pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+
+	// now need to iterate through the jsonb fields and add them
+	/* we don't expect nested values here */
+	it = JsonbIteratorInit(&sqlerrs_jsonb->root);
+	type = JsonbIteratorNext(&it, &val, true /*skip_nested*/);
+	if (type != WJB_BEGIN_OBJECT)
+		elog(ERROR, "invalid JSON format");
+	while ((type = JsonbIteratorNext(&it, &val, true)))
+	{
+		const char *errcode;
+		int64 errcnt;
+
+		if (type == WJB_END_OBJECT)
+			break;
+		else if (type == WJB_KEY)
+		{
+			errcode = pnstrdup(val.val.string.val, val.val.string.len);
+			// now get the value for this key
+			type = JsonbIteratorNext(&it, &val, true);
+			if (type != WJB_VALUE)
+				elog(ERROR, "unexpected jsonb type");
+			errcnt =
+				DatumGetInt64(DirectFunctionCall1(numeric_int8, NumericGetDatum(val.val.numeric)));
+			// now that we have both key and value, push them
+			ts_jsonb_add_int64(parse_state, errcode, errcnt);
+		}
+		else
+			// we are not expecting anything else for this
+			elog(ERROR, "unexpected jsonb type");
+	}
+	// close the jsonb that corresponds to this job_type
+	ret = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
+	return ret;
+}
+/* this function queries the database through SPI and gets back a set of records
+ that look like (job_type TEXT, jsonb_object_agg JSONB).
+ For example, (user_defined_action, {"P0001": 2, "42883": 5})
+ (we are expecting about 6 rows depending
+ on how we write the query and if we exclude any jobs)
+ Then for each returned row adds a new kv pair to the jsonb,
+ which looks like "job_type": {"errtype1": errcnt1, ...} */
+static void
+add_errors_by_sqlerrcode(JsonbParseState *parse_state)
+{
+	int res;
+	StringInfo command;
+	MemoryContext old_context = CurrentMemoryContext, spi_context;
+
+	const char *command_string =
+		"SELECT "
```

Comment:
cant we do `SET search_path TO pg_catalog, pg_temp` here and then run the query. Would make the query clearer and also safeguard against future non-qualified references in the query

Reply:
I had tried that suggestion initially but ran into some issues with the tests so did it like this eventually. I see your point though, I will try to do `SET search_path` instead

Reply:
Done